plugins/auth-backend/src/identity/router.ts

```diff
@@ -91,21 +93,30 @@ export function bindOidcRouter(
       );
     }
 
-    const { sub: userEntityRef, ent: ownershipEntityRefs = [] } =
-      decodeJwt(token);
+    const { sub: userEntityRef, ent: ownershipEntityRefs } = decodeJwt(token);
 
     if (typeof userEntityRef !== 'string') {
       throw new Error('Invalid user token, user entity ref must be a string');
     }
+
+    // Return user info if it's already available in the token (ie. it is a full token)
     if (
-      !Array.isArray(ownershipEntityRefs) ||
-      ownershipEntityRefs.some(ref => typeof ref !== 'string')
+      Array.isArray(ownershipEntityRefs) &&
+      ownershipEntityRefs.every(ref => typeof ref === 'string')
     ) {
-      throw new Error(
-        'Invalid user token, ownership entity refs must be an array of strings',
-      );
+      res.json({ sub: userEntityRef, ent: ownershipEntityRefs });
+      return;
     }
 
-    res.json({ sub: userEntityRef, ent: ownershipEntityRefs });
+    const userInfo = await userInfoDatabaseHandler.getUserInfo(userEntityRef);
+    if (!userInfo) {
+      res.status(404).send('User info not found');
+      return;
+    }
+
+    res.json({
+      sub: userInfo.userEntityRef,
+      ent: userInfo.ownershipEntityRefs,
```

Comment:
Regarding what's stored, and what's returned over the wire: Maybe it would simplify things if the storage model was basically `{ claims: JsonObject }`, so that what you find in the json column would typically be like `{"claims":{"sub":"user:default/freben","ent":[]}}` or so. At that point, maybe the returned body could have a `claims` root key too? 🤔

Maybe @Rugvip has opinions here, but I'm essentially wondering if who, if anyone, should do the translations back and forth between the long-form and short-form of each name an if it's at all worth it. Especially if we then open up for people to customize what set of claims that actually get persisted, so we can get extra stuff in here that come out of the sign in resolver. And with that `claims` key, we could also add the `profile` key next to it, with what came out of the profile transform during sign in.

So - what should the ideal response object shape be here?